src/nvim/option_vars.h

```diff
@@ -8,49 +8,43 @@
 // option_vars.h: definition of global variables for settable options
 
 // Option Flags
-#define P_ALLOCED      0x01U        ///< the option is in allocated memory,
-                                    ///< must use free_string_option() when
-                                    ///< assigning new value. Not set if default is
-                                    ///< the same.
-#define P_EXPAND       0x02U        ///< environment expansion.  NOTE: P_EXPAND can
+#define P_EXPAND       0x01U        ///< environment expansion.  NOTE: P_EXPAND can
```

Comment:
is it really required to bump all these flags or could we keep them as-is and just comment-out the P_ALLOCED line and explain that it's unused

Reply:
It's not required but it can be useful to free up flag space as we can only have up to 32 flags
Let me know if I should revert this change

Reply:
> plan to rework the whole option flag system in the future anyway to no longer use macros

it's probably worth avoiding the churn here, until then.

Reply:
Should be fine now